File: Graph_Traversal/1260_DFS와 BFS.cpp
#include <iostream>
#include <vector>
#include <algorithm> // sort

using namespace std;

int que[10001];
int fr = 0, re = 0;
vector<int> adj[1001]; //인접 리스트

void enqueue(int data) {
	que[re++] = data;
	return;
}

int dequeue(void) {
	return que[fr++];
}

// 깊이우선탐색
void DFS(vector<int>& visited, int cur) {
	visited[cur] = 1;
	cout << cur << ' '; // 출력

	for (int& next : adj[cur]) {
		if (visited[next] == 1)
			continue;

		DFS(visited, next); // 탐색
	}
	return;
}

// 너비우선탐색
void BFS(vector<int>& visited, int start) {
	visited[start] = 1;
	enqueue(start); // 삽입

	while (fr < re) {
		int cur = dequeue();
		cout << cur << ' ';

		for (int& next : adj[cur]) {
			if (visited[next] == 1)
				continue;

			visited[next] = 1;
			enqueue(next);
		}
	}
	return;
}

void solution(int n, int m, int v) {
	//정렬
	for (int i = 1; i <= n; i++)
		sort(adj[i].begin(), adj[i].end());
	vector<int> visited(n + 1, 0); // 방문리스트 선언

	// 먼저 DFS
	DFS(visited, v);
	cout << '\n';

	//초기화
	fill(visited.begin(), visited.end(), 0);
	BFS(visited, v);

	return;
}

int main(void) {
	int n, m, v;

	cin >> n >> m >> v;
	for (int i = 0; i < m; ++i) {
		int a, b;

		cin >> a >> b;
		adj[a].push_back(b);
		adj[b].push_back(a);
	}

	solution(n, m, v);

	return 0;
}

File: Graph_Traversal/1260_DFS와 BFS.py
from collections import deque
#입출력
from sys import stdin
#입력 
input = stdin.readline

#깊이우선탐색
def DFS(graph,v,visited):
    visited[v]=True
    print(v,end=' ')
    for i in graph[v]:
        if not visited[i]:
            DFS(graph,i,visited)

#너비우선탐색
def BFS(graph,v,visited):
    que=deque([v])
    visited[v]=True
    while que:
        v=que.popleft()
        print(v,end=' ')
        for i in graph[v]:
            if not visited[i]:
                que.append(i)
                visited[i]=True

n,m,v=map(int,input().split())
graph=[[] for _ in range(n+1)]
visited=[False]*(n+1)

for i in range(m):
    a,b=map(int,input().split())
    graph[a].append(b)
    graph[b].append(a)
    graph[a].sort()
    graph[b].sort()

DFS(graph,v,visited)
print()

visited=[False]*(n+1)
BFS(graph,v,visited)